src/adagrams.js

```diff
@@ -1,15 +1,190 @@
-export const drawLetters = () => {
-  // Implement this method for wave 1
+const letterCount = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
 };
 
-export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+const letterValue = {
+  A: 1,
+  B: 3,
+  C: 3,
+  D: 2,
+  E: 1,
+  F: 4,
+  G: 2,
+  H: 4,
+  I: 1,
+  J: 8,
+  K: 5,
+  L: 1,
+  M: 3,
+  N: 1,
+  O: 1,
+  P: 3,
+  Q: 10,
+  R: 1,
+  S: 1,
+  T: 1,
+  U: 1,
+  V: 4,
+  W: 4,
+  X: 8,
+  Y: 4,
+  Z: 10,
 };
 
-export const scoreWord = (word) => {
-  // Implement this method for wave 3
-};
+class Adagrams {
+  
+  static letterPool() {
+    // using the object.keys to get the array string of letters
+    let key = Object.keys(letterCount);
 
-export const highestScoreFrom = (words) => {
-  // Implement this method for wave 1
-};
+    // create the array of letters
+    let letterPool = [];
+
+    key.forEach((letter) => {
+      for (let i = 0; i < letterCount[letter]; i++) {
+        // getting the values how many times string letter appear in the array
+        // EX: X: 2 -> ['X', 'X]
+        letterPool.push(letter);
+      }
+    });
+    return letterPool;
+  };
+
+  static drawLetters() {
+    // create an empty array
+    let handLetters = [];
+    // create a copy and using 3 dot and new set method in array so we don't get the duplicate letters
+    let letterPoolCopy = [...new Set(Adagrams.letterPool())];
+    // using for to print 10 array of string letters
+    for (let i = 0; i < 10; i++) {
+      // set the copy length to the letters
+      let letters = letterPoolCopy.length;
+      let i = Math.floor(letters * Math.random());
+      // remove entry from letter_pool at index, add to letters array
+      handLetters.push(letterPoolCopy[i]);
+      letterPoolCopy.splice(i, 1);
+    }
+    return handLetters;
+  };
+
+  static usesAvailableLetters(input, lettersInHand) {
+    // using split to separate letter string input
+    const array = input.split("");
+    // create a copy of lettersInHand and also using the 3 dots
+    let lettersInHandCopy = [...lettersInHand];
```

Comment:
Consider scaling back on a few of these comments. The comments on line 96 and 98 don't add much clarity, they're just repeating exactly what the syntax says.